QKSMS/src/main/java/com/moez/QKSMS/mmssms/Transaction.java

```diff
@@ -510,7 +510,7 @@ public static MessageInfo getBytes(Context context, boolean saveMessage, String[
                 Log.e(TAG, "exception thrown", e);
 
                 // use the old way if something goes wrong with the persister
-                insert(context, recipients, parts, subject);
+                info.location = insert(context, recipients, parts, subject);
```

Comment:
This bit is a fallback if it can't use the other way to set info.location, and insert() is returning the location object, but it was not being assigned to anything. I'm pretty sure it's supposed to be assigning the return value to info.location, so when I was debugging my tests I added it in.